addon/components/basic-dropdown-content.ts

```diff
@@ -137,61 +137,65 @@ export default class BasicDropdownContent extends Component<Args> {
       this.scrollableAncestors = getScrollableAncestors(triggerElement);
     }
     this.addScrollHandling(dropdownElement);
-  }
-
-  @action
-  teardown(): void {
-    this.removeGlobalEvents();
-    this.removeScrollHandling();
-    this.scrollableAncestors = [];
-
-    document.removeEventListener(
-      this.args.rootEventType,
-      this.handleRootMouseDown as RootMouseDownHandler,
-      true
-    );
+    return () => {
+      this.removeGlobalEvents();
+      this.removeScrollHandling();
+      this.scrollableAncestors = [];
 
-    if (this.isTouchDevice) {
-      document.removeEventListener('touchstart', this.touchStartHandler, true);
       document.removeEventListener(
-        'touchend',
+        this.args.rootEventType,
         this.handleRootMouseDown as RootMouseDownHandler,
         true
       );
+
+      if (this.isTouchDevice) {
+        document.removeEventListener('touchstart', this.touchStartHandler, true);
+        document.removeEventListener(
+          'touchend',
+          this.handleRootMouseDown as RootMouseDownHandler,
+          true
+        );
+      }
     }
-  }
+  // @ts-ignore
+  }, { eager: false });
 
-  @action
-  animateIn(dropdownElement: Element): void {
-    if (!this.animationEnabled) return;
-    waitForAnimations(dropdownElement, () => {
-      this.animationClass = this.transitionedInClass;
+  initiallyReposition = modifier(() => {
+    // Escape autotracking frame and avoid backtracking re-render
+    Promise.resolve().then(() => {
+      this.args.dropdown.actions.reposition()
```

Comment:
This is the only code change I made. Since the `did-insert` modifier does not participate in auto tracking this was not an issue before.

I'm reasonably confident that this change is compatible since the tests pass. If this modifier doesn't run a number of tests will fail.

Someone more familiar with this library may have a better suggestion to avoid doing this operation with a derived data pattern (since it's a side-effect type modfifier).